### PR TITLE
Implements Content Event Mapping

### PR DESCRIPTION
**Pull request by ladanazita**
We interpret Chapters in Adobe as Content.
	Adobe also has a concept of Chapter Skipped,
	which Segment does not spec. We will be writing
	a proposal to spec this event, but for now
	will omit mapping this.

### CODE REVIEW

File: Example/Tests/Tests.m [hunk 1]
```diff
@@ -29,7 +29,7 @@ - (ADBMediaHeartbeat *)createWithDelegate:(id)delegate andConfig:(ADBMediaHeartb
```
Comment (ladanazita) [inline Example/Tests/Tests.m]: Add comment — why event type and what is it intended to do?

File: Pod/Classes/SEGAdobeIntegration.m [hunk 2]
```diff
@@ -24,12 +24,16 @@ - (ADBMediaHeartbeat *)createWithDelegate:(id)delegate andConfig:(ADBMediaHeartb
 
 @implementation SEGRealADBMediaObjectFactory
```
Comment (ladanazita) [inline Pod/Classes/SEGAdobeIntegration.m]: Add comment — why event type and what is it intended to do?

File: Pod/Classes/SEGAdobeIntegration.m [hunk 3]
```diff
@@ -24,12 +24,30 @@ - (ADBMediaHeartbeat *)createWithDelegate:(id)delegate andConfig:(ADBMediaHeartb
...
 
-- (ADBMediaObject *_Nullable)createWithProperties:(NSDictionary *_Nullable)properties;
+
+/**
+ Creates an instance of ADBMediaObject to pass through
+ Adobe Heartbeat Events.
+
+ ADBMediaObject can build a Video, Chapter, or Ad object.
+ Passing in a value for event type as Playback, Content, or
+ Ad will build the relevant instance of ADBMediaObject,
```
Comment (ladanazita) [inline Pod/Classes/SEGAdobeIntegration.m]: Ad will come once I implement Ad tracking

File: Pod/Classes/SEGAdobeIntegration.m [hunk 4]
```diff
@@ -398,6 +423,19 @@ - (void)trackHeartbeatEvents:(SEGTrackPayload *)payload
         return;
     }
 
+    if ([payload.event isEqualToString:@"Video Content Started"]) {
+        self.mediaObject = [self createMediaObject:payload.properties andEventType:@"Content"];
+        [self.ADBMediaHeartbeat trackEvent:ADBMediaHeartbeatEventChapterStart mediaObject:self.mediaObject data:payload.properties];
```
Comment (ladanazita) [inline Pod/Classes/SEGAdobeIntegration.m]: Should also trigger `trackPlay`

File: Pod/Classes/SEGAdobeIntegration.m [hunk 5]
```diff
@@ -398,6 +423,26 @@ - (void)trackHeartbeatEvents:(SEGTrackPayload *)payload
...
+        SEGLog(@"[ADBMediaHeartbeat trackEvent:ADBMediaHeartbeatEventChapterStart mediaObject:%@ data:%@]", self.mediaObject, payload.properties);
+        return;
+    }
+
+    if ([payload.event isEqualToString:@"Video Content Completed"]) {
+        [self.ADBMediaHeartbeat trackComplete];
+        SEGLog(@"[ADBMediaHeartbeat trackComplete]");
+
+        self.mediaObject = [self createMediaObject:payload.properties andEventType:@"Content"];
+        [self.ADBMediaHeartbeat trackEvent:ADBMediaHeartbeatEventChapterComplete mediaObject:self.mediaObject data:payload.properties];
```
Comment (ladanazita) [inline Pod/Classes/SEGAdobeIntegration.m]: @brennan this differs from the [Android implementation here](https://github.com/segment-integrations/analytics-android-integration-adobe-analytics/blob/master/src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java#L565) I think we should consider also mapping a `trackEvent(MediaHeartbeat.Event.ChapterComplete, MediaObject, Map<String, String> metadata) ` in Android here — thoughts?

Comment (brennan) [reply Pod/Classes/SEGAdobeIntegration.m]: Yes, I agree that we should add a `ChapterComplete` event - I can update this in the Android library. That said, it should probably be `heartbeat.trackEvent(MediaHeartbeat.Event.ChapterComplete, null, null);` - this is how Adobe implements `ChapterComplete` in their examples.

Comment (ladanazita) [reply Pod/Classes/SEGAdobeIntegration.m]: Hmm yeah I see that [in iOS too](https://github.com/Adobe-Marketing-Cloud/video-heartbeat-v2/blob/master/sdks/iOS/samples/BasicPlayerSample/BasicPlayerSample/Classes/analytics/VideoAnalyticsProvider.m#L158).

I will change now

**Pull request merged**